hack/update_submodules.sh

```diff
@@ -123,30 +123,25 @@ if [ ! -z "$remote_branch" ]; then
   fi
 fi
 
-# Return if the working branch is dirty
-if ! git diff --quiet; then
-  echo "branch dirty, exiting to not overwrite work"
-  exit 1
-fi
 
-# For each branch that the user gives (master, release-4.7), create a new branch based off it,
-# with commits updating the submodules
+WMCO_ROOT=$(dirname "${BASH_SOURCE}")/..
+source $WMCO_ROOT/hack/common.sh
+
+# Check if branch is dirty, and exit if it is
+check_git_branch
```

Comment:
If the function returns 1 does it automatically exit here? From what I understand, return exits terminates the subshell. Don't you have to do some return value checking here?

Reply:
because we set the -e at the top, the return exits the subshell, and then since a command exited with a non-zero, the -e exits the program.